src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -177,6 +177,37 @@ public void UpdateHistoryLog(ClusterOperationContext context, long index, long t
             UpdateInternal(context, cmd, guid, type, index, term, HistoryStatus.Committed, result, exception);
         }
 
+        public void UpdateHistoryLogPreservingGuidAndStatus(ClusterOperationContext context, long index, long term, BlittableJsonReaderObject cmd, object result, Exception exception)
```

Comment:
can you explain the logic here?

Reply:
The command we want to remove doesn't actually being removed, but replaced with `noop` with the same guid and state.
We are preserving the guide because we cant update a command without having it (its guid) in the history log.